ntbs-service/Models/Entities/Notification.cs

```diff
@@ -114,11 +115,21 @@ public bool ShouldBeClosed()
         {
             var mostRecentTreatmentEvent = TreatmentEvents.GetMostRecentTreatmentEvent();
 
-            return NotificationStatus == NotificationStatus.Notified
-                && mostRecentTreatmentEvent != null
-                && mostRecentTreatmentEvent.TreatmentEventTypeIsOutcome
-                && mostRecentTreatmentEvent.TreatmentOutcome?.TreatmentOutcomeSubType != TreatmentOutcomeSubType.StillOnTreatment
-                && mostRecentTreatmentEvent.EventDate < DateTime.Today.AddYears(-1);
+            if (NotificationStatus == NotificationStatus.Notified
+                && mostRecentTreatmentEvent?.TreatmentOutcomeId != null
+                && mostRecentTreatmentEvent.TreatmentEventTypeIsOutcome)
+            {
+                if (mostRecentTreatmentEvent.TreatmentOutcome == null)
+                {
+                    throw new ApplicationException("ShouldBeClosed cannot be called without loaded outcome events.");
+                }
+
+                return mostRecentTreatmentEvent.TreatmentOutcome != null
+                       && mostRecentTreatmentEvent.TreatmentOutcome.TreatmentOutcomeSubType != TreatmentOutcomeSubType.StillOnTreatment
+                       && mostRecentTreatmentEvent.EventDate < DateTime.Today.AddYears(-1);
```

Comment:
To remove the nesting in this method, perhaps we should pull this out into a local function?